pkg/kafka/producer/config.go

```diff
@@ -42,6 +46,8 @@ func (c *Configuration) NewProducer() (sarama.AsyncProducer, error) {
 	saramaConfig.Producer.Compression = c.Compression
 	saramaConfig.Producer.CompressionLevel = c.CompressionLevel
 	saramaConfig.Producer.Return.Successes = true
+	saramaConfig.Producer.Flush.Bytes = c.BatchSize
+	saramaConfig.Producer.Flush.Frequency = c.BatchLinger
```

Comment:
we might as well add BatchMaxMessages for completeness.

Reply:
Added BatchMaxMessages and set all batch defaults to 0.